apps/web/src/studio/components/workflows/step-editor/WorkflowStepEditorControlsPanel.tsx

```diff
@@ -34,8 +34,12 @@ export const WorkflowStepEditorControlsPanel: FC<IWorkflowStepEditorControlsPane
   const { Component, toggle, setPath } = useDocsModal();
   const havePayloadProperties = useMemo(() => {
     return (
-      Object.keys(workflow?.payload?.schema || workflow?.options?.payloadSchema || workflow?.payloadSchema || {})
-        .length > 0
+      Object.keys(
+        workflow?.payload?.schema?.properties ||
+          workflow?.options?.payloadSchema?.properties ||
+          workflow?.payloadSchema?.properties ||
+          {}
+      ).length > 0
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
✏ **suggestion:** Consider extracting to a selector (reusable pure function that takes a workfl‏ow and returns this value)